src/App.tsx

```diff
@@ -1,56 +1,102 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/peopleList';
+
+const filterPeople = (query: string) => {
+  const normalizedQuery = query.trim().toLowerCase();
+
+  return peopleFromServer.filter(person => {
+    const normalizedName = person.name.trim().toLowerCase();
+
+    return normalizedName.includes(normalizedQuery);
+  });
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [delay] = useState(1000);
+  const [selectedPerson, setSelectedPerson] = useState<null | Person>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery((event.target.value).trim());
+    setIsDropdownActive(true);
+  };
+
+  const preparedPeople = useMemo(
+    () => filterPeople(appliedQuery),
+    [appliedQuery],
+  );
+
+  const handlePersonClick = (person: Person) => {
+    setQuery(person.name);
+    setSelectedPerson(person);
+    setIsDropdownActive(false);
+    setAppliedQuery('');
+  };
+
+  const [isInputFocused, setIsInputFocused] = useState(false);
+
+  const handleInputFocus = () => {
+    setIsInputFocused(true);
+  };
+
+  const handleInputBlur = () => {
+    setIsInputFocused(false);
+    // setAppliedQuery('');
```

Comment:
The line 'setAppliedQuery('');' is commented out. If this line is not needed, consider removing it to clean up the code.